### PR TITLE
Lots Here!

### PR DESCRIPTION
**Pull request by Jeff-Moreau**
Puzzles
Meshes
Colliders
Blocking Volumes
Hub Update
Level One Remake
Blueprints
C++
God knows what Else, I am sure i missed some things to clean up. But i can do that as we go.. This is just getting very large and would like to get this in the game for all to use

### CODE REVIEW

File: Source/SteveSloth/Private/MyValve.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{
+	Super::Tick(DeltaTime);
+	
+	if (isInArea)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Player in Area"));
+	}
+}
+
+void AMyValve::OnHitboxOverlapBegin(UPrimitiveComponent* OverlappedComp, AActor* OtherActor, UPrimitiveComponent* OtherComp,
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyValve.cpp:46]: right now this isnt being called, is it necessary

File: Source/SteveSloth/Public/MyPlayer.h [hunk 2]
```diff
@@ -118,24 +119,33 @@ class STEVESLOTH_API AMyPlayer : public ACharacter
...
 
+public: // GETTERS/ACCESSORS
+	float GetMaxHealth() { return MaxHealth; }
+	float GetCurrentHealth() { return CurrentHealth; }
+	int GetEquippedAmmoIndex() { return EquippedAmmoIndex; }
+
+public: // SETTERS/MUTATORS
+	void SetMaxHealth(float amount) { MaxHealth = MaxHealth + amount; }
+	void SetCurrentHealth(float amount) { CurrentHealth = CurrentHealth + amount; }
+
```
Comment (Saretzky) [inline Source/SteveSloth/Public/MyPlayer.h:130]: why are these moved here? shouldnt the getters & setters be right above the constructor

File: Source/SteveSloth/Private/MyEnemyManager.cpp [hunk 3]
```diff
@@ -18,7 +18,24 @@ AMyEnemyManager::AMyEnemyManager()
...
+	if (levelName == "LevelOne")
+	{
+		for (int i = 0; i < SpawnPoints.Num(); i++)
+		{
+			FVector SpawnLocation = SpawnPoints[i]->GetTransform().GetLocation();
+			FRotator SpawnRotation = SpawnPoints[i]->GetTransform().GetRotation().Rotator();
+
+			AActor* tryThis = GetWorld()->SpawnActor<AActor>(theKoala, SpawnLocation, SpawnRotation, spawnParams);
+		}
+	}
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyEnemyManager.cpp:38]: I assume we'll have to add all the other enemies in the same way?

**Review by Saretzky — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Private/MyEnemyManager.cpp (hunk 3 above)
Comment (Jeff-Moreau) [reply Source/SteveSloth/Private/MyEnemyManager.cpp:38]: I did not add this lol.. funny.. dont know who did

File: Source/SteveSloth/Public/MyPlayer.h (hunk 2 above)
Comment (Jeff-Moreau) [reply Source/SteveSloth/Public/MyPlayer.h:130]: this is because i messed up when merging from main.. ill bring them back to where they belong

File: Source/SteveSloth/Private/MyValve.cpp (hunk 1 above)
Comment (Jeff-Moreau) [reply Source/SteveSloth/Private/MyValve.cpp:46]: yes this is not used.. ill remove

File: Source/SteveSloth/Private/MyEnemyManager.cpp (hunk 3 above)
Comment (Saretzky) [reply Source/SteveSloth/Private/MyEnemyManager.cpp:38]: this whole section?

Comment (Jeff-Moreau) [reply Source/SteveSloth/Private/MyEnemyManager.cpp:38]: nevermind i know what this is.. was going to be a list then i abandoned for now until enemies are done

Comment (Saretzky) [reply Source/SteveSloth/Private/MyEnemyManager.cpp:38]: i see, so should we keep this for now?

File: Content/Artwork/Puzzles/uploads_files_3652793_triangle_hanging_light.uasset
Comment (Saretzky) [inline Content/Artwork/Puzzles/uploads_files_3652793_triangle_hanging_light.uasset:1]: rename (remove uploads_files_3652793)

**Review by Saretzky — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Public/MyLevelManager.h
Comment (Saretzky) [inline Source/SteveSloth/Public/MyLevelManager.h:1]: we decided to name bools with a lowercase b at the beginning

File: Source/SteveSloth/Public/MyValve.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	UStaticMeshComponent* ValveMesh;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite)
+	UBoxComponent* PlayerHitBox;
+
+private: // PRIVATE INTERNAL VARIABLES
+	AMyPlayer* Steve;
+	ACharacter* Player;
+
+	bool isInArea;
```
Comment (Saretzky) [inline Source/SteveSloth/Public/MyValve.h:41]: can remove this now

File: Source/SteveSloth/Private/MyValve.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+AMyValve::AMyValve()
+{
+	PrimaryActorTick.bCanEverTick = true;
+
+	BaseMesh = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("Base"));
+	ValveMesh = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("Valve"));
+	PlayerHitBox = CreateDefaultSubobject<UBoxComponent>(TEXT("Player Area"));
+
+	isInArea = false;
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyValve.cpp:21]: remove this too

File: Content/Artwork/Puzzles/uploads_files_3652793_triangle_hanging_light.uasset
Comment (Jeff-Moreau) [reply Content/Artwork/Puzzles/uploads_files_3652793_triangle_hanging_light.uasset:1]: done

**Review by Saretzky — CHANGES_REQUESTED** [review]
small changes

**Review by Saretzky — APPROVED** [review]
looks good

**Review by CrazyTobi — APPROVED** [review]
Looking good
